### DISCUSSION TITLE
GUI Elements

### DISCUSSION THREAD

**Discussion by benjaminrwilson** [datoviz/datoviz#21] [Q&A] [answered]
I had two questions regarding GUI elements.

1. Is there a way to add a text field as a GUI component?
2. Additionally, is it possible to have a "dropdown menu"? e.g., a listing of files in a directory.

Thanks!

**Comment by rossant** [comment, marked as answer]
1. Editable text box? There's `dvz_gui_textbox()`
2. Yes I'd like to add a dropdown menu asap.

Adding new controls is just a matter of wrapping them properly in the Datoviz C API. There's a [checklist for developers here](https://datoviz.org/discussions/checklists/#implementing-a-new-gui-control). The functionality is provided by [Dear ImGui](https://github.com/ocornut/imgui) which as a huge library of rich GUI controls.

**Comment by rossant** [comment]
Text box and dropdown are available in Datoviz v0.3.
